lib/Fhp/Model/TanRequestChallengeImage.php

```diff
@@ -25,15 +25,33 @@ public function __construct(Bin $bin)
         $mimeTypeLength = ord($mimeTypeLengthString[0]) * 256 + ord($mimeTypeLengthString[1]);
 
         $this->mimeType = substr($data, 2, $mimeTypeLength);
-        $this->data = substr($data, 2 + $mimeTypeLength + 2);
+
+        $data = substr($data, 2 + $mimeTypeLength);
+
+        $dataLengthString = substr($data, 0, 2);
+        $expectedDataLength = ord($dataLengthString[0]) * 256 + ord($dataLengthString[1]);
+        $isDataLength = strlen($data) - 2;
```

Comment:
nit: "actualDataLength" -- one would expect "is" to be a bool.

Reply:
You are right. Changed.